src/background/interpreter.ts

```diff
@@ -13,11 +13,16 @@ export class Interpreter {
         this.messageToActionMapping[messageType] = callback;
     };
 
-    public interpret(message: InterpreterMessage): boolean {
+    public interpret(message: InterpreterMessage): InterpreterResponse {
         if (this.messageToActionMapping[message.messageType]) {
-            this.messageToActionMapping[message.messageType](message.payload, message.tabId);
-            return true;
+            return {
+                success: true,
+                result: this.messageToActionMapping[message.messageType](
+                    message.payload,
+                    message.tabId,
+                ),
+            };
         }
-        return false;
+        return { success: false };
```

Comment:
When I was reading the usage of this property in the message distributor, the term `success` felt a little ambiguous to me - it wasn't clear whether it implied just "this interpreter handled this message vs not" or "the handler for this message worked without error vs not". If it's strictly the former, maybe `handledMessage` or something would read more clearly?